cmd/xl-storage.go

```diff
@@ -498,7 +498,7 @@ func (s *xlStorage) NSScanner(ctx context.Context, cache dataUsageCache, updates
 			atomic.AddUint64(&globalScannerStats.accTotalVersions, 1)
 			oi := version.ToObjectInfo(item.bucket, item.objectPath())
 			sz := item.applyActions(ctx, objAPI, oi, &sizeS)
-			if !oi.DeleteMarker && sz == oi.Size {
+			if oi.VersionID != "" && sz == oi.Size {
```

Comment:
> This looks like something new @vadmeste

Yes, I thought if we are counting versions, we should count the delete makers as well. We should not omit it.

Reply:
hmm yes but we can separate them? @vadmeste since DeleteMarkers are usually the culprit?

Reply:
> hmm yes but we can separate them? @vadmeste since DeleteMarkers are usually the culprit?

yes, we can do that - but this requires more involved work. I will do it if you think it is not enough to analyze the ratio versions/objects of the overall cluster to detect if something is wrong.